### DISCUSSION TITLE
v4 requires all pages to export metadata

### DISCUSSION THREAD

**Discussion by ArikSquad** [shuding/nextra#3828] [General]
If I have a CSR page, it will error out, saying that 'metadata' is not exported. And the thing about them, is that they can't export metadata. I currently have added metadata to SSR pages.

What is even the reason to find the metadata of every single page than just the ones needed? (useContentDir and subdirectory on docs?). It might be a possibility that this is not even nextra's fault, but then, why doesn't it respect layouts, but in nextras case it does make sense. I have my common metadata there and I wouldn't like to make all my client pages a serverside one with a client component (or is this the standard)

_This message is really messy, I hope you understand what I mean though_

**Comment by dimaMachina** [comment]
for now, it's nextra requirement for all pages `mdx`/`tsx`/`jsx` to have exported `metadata`, it will be maybe changed after 4 rc.0 release, after users feedback

**Reply by ftzi** [reply]
@dimaMachina I want only my `/docs` route to use Nextra. It's pretty bad that my non-Nextra routes are requiring to export metadata.

Can't we only require metadata to routes related to Nextra?

**Reply by codenamethanos** [reply]
I also want Nextra to not require metadata export in all page.js files. I have layout.js with metadata export and CSR pages where we cannot export metadata. In both cases Nextra 4 throws an error.

**Comment by clicktodev** [comment]
I just found about this, @dimaMachina any plans to address this soon?

**Comment by dimaMachina** [comment]
fixed in https://github.com/shuding/nextra/releases/tag/nextra%404.2.3

**Reply by ftzi** [reply]
Awesome!!!!
